### PR TITLE
PCHR-1432: Public vacancy list redesign

### PR DESCRIPTION
**Pull request by brunoqueiros**
This PR overrides public vacancy list page styles to look as close to CiviHR design. It was created a style for striped table and adjusted some spacings.

***Before***
image

image

***After***
image

image

### CODE REVIEW

**Comment by JonScreat** [comment]
Hi, so I'm not altogether sure what I'm looking at here.

I believe this is a Self Service Portal page, correct? Only it looks like an Admin page, and I can see in the ticket that you have been linked to the Admin styleguide.

We did have this page designed for SSP Vacancies: https://invis.io/W74KSD5RY#/108448353_Job-Vacancies---V5a but that's not been referenced in the ticket.

**Comment by AkA84** [comment]
@brunoqueiros can you make sure to screengrab the whole page instead of cropping the screenshots just on the area that changed? This way the context is clear

**Comment by JonScreat** [comment]
**Vacancy List**

- Table should have a white background (no banding) with 1px #e8eef0 lines. Table headings (first row) should be Open Sans Bold 13px in #4d5663 (This is actually a change to tables throughout CiviHR to remove he banding which will be addressed in the next style guide)

- Table should also have 1px #e8eef0 outline and 2px rounded corners.

**Vacancy**

- Same points with colour, outline and rounded corners for the table.

- Background (including header section) should be #f4f7f8, with the header section separated with a 1px #e8eef0 line

- Header/job title should be Open Sans Semibold in #4d5663. There is also a CiviCRM logo here which should be removed.

- The second ‘job title’ with the underline should be removed.

- In the table the labels should be Open Sans Semibold (#4d5663) and the copy should be Open Sans Regular (#727e8a).

- The pop-up itself should not have rounded corners.

- The close button looks oddly shaped? This should also be further away from the edge (20px)

- The bottom of the popup should be 20px below the apply and lose buttons so there is no empty space underneath.

**Comment by brunoqueiros** [comment]
@JonScreat I updated the screenshots.

**Comment by JonScreat** [comment]
**Vacancy List**

- Does the table have a shadow? Hard to tell from the screenshot, but it should have a shadow the same as the page title above. 

- Copy in the table (e.g. ‘headquarters’ should be (#727e8a)

**Vacancy**

- Looks like the pop-up has a boarder on the screenshot? This should be removed. Also there look to be a few black dots in the bottom right corner?

File: org.civicrm.bootstrapcivicrm/scss/bootstrap/bootstrap.scss [hunk 1]
```diff
@@ -5,14 +5,14 @@
...
+  @import "vendor/ui-bootstrap/ui-bootstrap";
+  @import "vendor/ui-select/ui-select";
 
-    @import "overrides/style/*";
-    @import "components/*";
+  @import "overrides/style/*";
+  @import "components/*";
 }
 
+@import "vendor/text-angular/text-angular";
```
Comment (AkA84) [inline org.civicrm.bootstrapcivicrm/scss/bootstrap/bootstrap.scss]: Why did you move it outside the namespace? Are you sure this does not affect any other pages where text-angular is used?

File: org.civicrm.bootstrapcivicrm/scss/civicrm/common/_buttons.scss [hunk 2]
```diff
@@ -12,7 +12,7 @@ $button-border-radius: 2px;
     opacity: 1 !important;
     font-size: 1em !important;
 
-    &:not(.crm-form-submit):not(.crm-close-accordion):not(.crm-clear-link) {
+    &:not(.crm-form-submit):not(.crm-close-accordion):not(.crm-clear-link):not([href^="/index.php?q=civicrm/vacancy/apply&reset=1&id="]) {
```
Comment (AkA84) [inline org.civicrm.bootstrapcivicrm/scss/civicrm/common/_buttons.scss]: This is way over-specific. What are you trying to accomplish here? Isn't there a better selector to do it? In any case i think this selector in general deserves a comment to explain why it has this `:not()`s in the first place

File: org.civicrm.bootstrapcivicrm/scss/civicrm/vacancies/_vacancies.scss [hunk 3]
```diff
@@ -141,4 +141,116 @@
       }
     }
   }
+
+  &.page-civicrm-vacancy-publiclisting table {
+    @extend %crm-table;
```
Comment (AkA84) [inline org.civicrm.bootstrapcivicrm/scss/civicrm/vacancies/_vacancies.scss]: Why this styles are applied only on this particular table? I presume `%crm-table` contains the style that all "old civi" tables should have? If so, do you think this styles here should go in that placeholder? Or is this table an exception?

File: org.civicrm.bootstrapcivicrm/scss/civicrm/vacancies/_vacancies.scss [hunk 4]
```diff
@@ -141,4 +141,116 @@
...
+      td {
+        color: $crm-copy;
+      }
+    }
+
+    &.vacancy-popup {
+      margin-bottom: 20px;
+    }
+
+    &.selector {
```
Comment (AkA84) [inline org.civicrm.bootstrapcivicrm/scss/civicrm/vacancies/_vacancies.scss]: What's this `.selector`?

File: org.civicrm.bootstrapcivicrm/scss/bootstrap/bootstrap.scss (hunk 1 above)
Comment (brunoqueiros) [inline org.civicrm.bootstrapcivicrm/scss/bootstrap/bootstrap.scss]: Because it was showing the `input` and this import has the style to hide it. The other pages that already use text-angular will continue working.

image

File: org.civicrm.bootstrapcivicrm/scss/civicrm/vacancies/_vacancies.scss [hunk 5]
```diff
@@ -141,4 +141,116 @@
...
+      margin-top: 20px;
+    }
+  }
+
+  .crm-container.ui-dialog {
```
Comment (AkA84) [inline org.civicrm.bootstrapcivicrm/scss/civicrm/vacancies/_vacancies.scss]: Is this dialog a generic civicrm dialog? If yes, do you think it could deserve a standalone partials with generic rules (right now this styling is applied only for this page), so that modals (i assume this is for a modal) gets automatically styled in every page?

File: org.civicrm.bootstrapcivicrm/scss/civicrm/vacancies/_vacancies.scss [hunk 6]
```diff
@@ -141,4 +141,116 @@
...
+    border-radius: 0;
+    background: $crm-table-striped-bg;
+
+    .ui-dialog-titlebar {
+      background: $crm-table-striped-bg !important;
+      border-bottom: 1px solid $crm-background !important;
+      border-radius: 0 !important;
+
+      > .ui-button {
+        &:nth-child(2) {
```
Comment (AkA84) [inline org.civicrm.bootstrapcivicrm/scss/civicrm/vacancies/_vacancies.scss]: Usual thing: why an index-based styling? There's no better way to do this?

File: org.civicrm.bootstrapcivicrm/scss/bootstrap/bootstrap.scss (hunk 1 above)
Comment (AkA84) [inline org.civicrm.bootstrapcivicrm/scss/bootstrap/bootstrap.scss]: Please check it in the Appraisals extensions as well (you probably need to install the extension on your local site). I believe text-angular is used in one of the modals

File: org.civicrm.bootstrapcivicrm/scss/civicrm/common/_buttons.scss [hunk 7]
```diff
@@ -12,7 +12,7 @@ $button-border-radius: 2px;
     opacity: 1 !important;
     font-size: 1em !important;
 
-    &:not(.crm-form-submit):not(.crm-close-accordion):not(.crm-clear-link) {
+    &:not(.crm-form-submit):not(.crm-close-accordion):not(.crm-clear-link):not(.ui-dialog .button) {
```
Comment (AkA84) [inline org.civicrm.bootstrapcivicrm/scss/civicrm/common/_buttons.scss]: Please add a comment explaining what this entire selector does

File: org.civicrm.bootstrapcivicrm/scss/civicrm/common/_modals.scss [hunk 8]
```diff
@@ -1,29 +1,79 @@
...
     padding: 20px;
     background: none !important;
     border: none !important;
+    border-bottom: 1px solid $crm-background !important;
     color: $gray-darker !important;
 
     .ui-dialog-title {
       @extend .modal-title, %open-sans-font-family;
       font-size: 18px;
+
```
Comment (AkA84) [inline org.civicrm.bootstrapcivicrm/scss/civicrm/common/_modals.scss]: empty line

File: org.civicrm.bootstrapcivicrm/scss/civicrm/vacancies/_vacancies.scss [hunk 9]
```diff
@@ -141,4 +141,49 @@
       }
     }
   }
+
+  table {
```
Comment (AkA84) [inline org.civicrm.bootstrapcivicrm/scss/civicrm/vacancies/_vacancies.scss]: Maybe I wasn't clear before, my point was if this table styling should be applied to *all* tables or if it's just for this specific table for this specific page. If it's for all tables, then we might need to move it in a separate partial and outside the `.page-civicrm-vacancy` namespace

**Comment by AkA84** [comment]
Approved
